packages/react-native/Libraries/AppDelegate/RCTRootViewFactory.mm

```diff
@@ -241,11 +241,35 @@ - (void)didCreateContextContainer:(std::shared_ptr<facebook::react::ContextConta
   contextContainer->insert("ReactNativeConfig", _reactNativeConfig);
 }
 
+- (NSArray<id<RCTBridgeModule>> *)extraModulesForBridge:(RCTBridge *)bridge {
+  if (_configuration.extraModulesForBridge != nil) {
+    return _configuration.extraModulesForBridge(bridge);
+  }
+  return nil;
+}
+
+- (NSDictionary<NSString *,Class> *)extraLazyModuleClassesForBridge:(RCTBridge *)bridge {
+  if (_configuration.extraLazyModuleClassesForBridge != nil) {
+    return _configuration.extraLazyModuleClassesForBridge(bridge);
+  }
+  return nil;
+}
+
 - (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
 {
+  if (_configuration.sourceURLForBridge != nil) {
+    return _configuration.sourceURLForBridge(bridge);
+  }
   return [self bundleURL];
```

Comment:
i would prefer not supporting `sourceURLForBridge:`. otherwise the logic for RCTRootViewFactory is diverse from sourceURLForBridge: and bundleURL. would be good to keep bundleURL as a single source of truth.
since we now route sourceURLForBridge: to bundleURL in https://github.com/facebook/react-native/blob/1c52d385de662f034adf05df37073681cf609d78/packages/react-native/template/ios/HelloWorld/AppDelegate.mm#L17-L29
if people follow the same code, it should be good to just keep bundleURL.

Reply:
Technically yes, but I think `sourceURLForBridge:` should still be called for old arch - maybe it's time to deprecate it (?) I'm not sure

Reply:
not 100% sure either. alright let's keep it as-is.

Reply:
yea so essentially sourceURLForBridge is the legacy API and bundleURL is the new one, technically at this point you can just override bundleURL and things should still work, but i think this is fine for consistency's sake